src/backend/distributed/shardsplit/cdc_decoder.c

```diff
@@ -0,0 +1,388 @@
+/*-------------------------------------------------------------------------
+ *
+ * cdc_decoder.c
+ *		CDC Decoder plugin for Citus
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "common/hashfn.h"
+#include "utils/typcache.h"
+#include "utils/lsyscache.h"
+#include "catalog/pg_namespace.h"
+#include "distributed/cdc_decoder.h"
+#include "distributed/relay_utility.h"
+#include "distributed/worker_protocol.h"
+#include "distributed/metadata_cache.h"
+
+static LogicalDecodeChangeCB ouputPluginChangeCB;
+
+
+static bool replication_origin_filter_cb(LogicalDecodingContext *ctx, RepOriginId
+										 origin_id);
+
+static void TranslateChangesIfSchemaChanged(Relation relation, Relation targetRelation,
+											ReorderBufferChange *change);
+
+static void TranslateAndPublishRelationForCDC(LogicalDecodingContext *ctx,
+											  ReorderBufferTXN *txn,
+											  Relation relation,
+											  ReorderBufferChange *change, Oid shardId,
+											  Oid targetRelationid);
+
+typedef struct
+{
+	Oid shardId;
+	Oid distributedTableId;
```

Comment:
After an alter_distributed_table we're no longer seeing inserts, presumably because the distributedTableId no longer exists. Not sure how we should deal with that. We should maybe punt for now.

Reply:
Maybe `SetupReplicationOriginLocalSession` need to go into `ConvertTable` itself so that we cover all different conversions?

Reply:
alter_distribute_table changes the Oid of the distributed table, so the CDC client's mapping is no longer valid and hence any inserts/update/deletes after that, is not able to be applied by the CDC subscriber.  The only current option is for the subscriber to drop the subscription and recreate the subscription again. This will be documented as a known issue.